pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/AvroRecordToPinotRowGenerator.java

```diff
@@ -32,15 +33,30 @@
 
 public class AvroRecordToPinotRowGenerator {
   private final Schema indexingSchema;
+  private final String timeColumnName;
+  private final FieldSpec timeFieldSpec;
 
   public AvroRecordToPinotRowGenerator(Schema indexingSchema) {
     this.indexingSchema = indexingSchema;
+    timeColumnName = indexingSchema.getTimeFieldSpec().getIncomingTimeColumnName();
+    if (indexingSchema.getTimeColumnName() != timeColumnName) {
```

Comment:
You need to use `.equals()` to check for equality/inequality in Java, otherwise it just compares the reference, which is not what you want here.